optimade/validator/utils.py

```diff
@@ -365,15 +361,15 @@ def wrapper(
                     success_type = "optional" if optional else None
                     validator.results.add_success(f"{request} - {msg}", success_type)
             else:
-                request = request.replace("\n", "")
+                request = request.replace("\n", "")  # type: ignore[union-attr]
                 message = msg.split("\n")
                 if validator.verbosity > 1:
                     # ValidationErrors from pydantic already include very detailed errors
                     # that get duplicated in the traceback
                     if not isinstance(result, ValidationError):
                         message += traceback.split("\n")
 
-                message = "\n".join(message)
+                message = "\n".join(message)  # type: ignore[assignment]
```

Comment:
You get an error here because you store a string in a variable `message` that has the type list.
One way to solve this would be to replace `message` on line 384 with  `"\n".join(message)` and delete line 372.

I d